packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -1249,15 +1252,30 @@ async function handler(
         },
       }),
     };
-    if (req.body.allRecurringDates) {
-      if (req.body.isFirstRecurringSlot) {
-        for (
-          let i = 0;
-          i < req.body.allRecurringDates.length && i < req.body.numSlotsToCheckForAvailability;
-          i++
-        ) {
-          const start = req.body.allRecurringDates[i].start;
-          const end = req.body.allRecurringDates[i].end;
+    if (req.body.allRecurringDates && req.body.isFirstRecurringSlot) {
+      for (
+        let i = 0;
+        i < req.body.allRecurringDates.length && i < req.body.numSlotsToCheckForAvailability;
+        i++
+      ) {
+        const start = req.body.allRecurringDates[i].start;
+        const end = req.body.allRecurringDates[i].end;
+        if (eventType.schedulingType == SchedulingType.ROUND_ROBIN) {
+          const fixedUsers = eventTypeWithUsers.users.filter((user: IsFixedAwareUser) => user.isFixed);
+          // each fixed user must be available
+          for (const key in fixedUsers) {
+            await ensureAvailableUsers(
+              { ...eventTypeWithUsers, users: [fixedUsers[key]] },
```

Comment:
we are repeating filtering step inside for loop 

```
const fixedUsers = eventTypeWithUsers.users.filter((user: IsFixedAwareUser) => user.isFixed)
```

we can move fixedUsers above for loop